internal/services/network/network_watcher_flow_log_resource.go

```diff
@@ -389,12 +415,18 @@ func resourceNetworkWatcherFlowLogRead(d *pluginsdk.ResourceData, meta interface
 				d.Set("storage_account_id", props.StorageId)
 			}
 
-			networkSecurityGroupId := ""
-			nsgId, err := networksecuritygroups.ParseNetworkSecurityGroupIDInsensitively(props.TargetResourceId)
-			if err == nil {
-				networkSecurityGroupId = nsgId.ID()
+			targetResourceId := props.TargetResourceId
+			if nsgId, err := networksecuritygroups.ParseNetworkSecurityGroupIDInsensitively(props.TargetResourceId); err == nil {
+				targetResourceId = nsgId.ID()
+			} else if vnetId, err := commonids.ParseVirtualNetworkIDInsensitively(props.TargetResourceId); err == nil {
+				targetResourceId = vnetId.ID()
+			}
+
+			if !features.FivePointOhBeta() {
+				d.Set("network_security_group_id", targetResourceId)
 			}
```

Comment:
With this logic we would potentially be setting a Virtual Network ID into `network_security_group_id`, I wonder whether we should only be setting this if the target resource ID is a NSG?